### DISCUSSION TITLE
TypeScript icon missing with new exports/types field (ts 4.7)

### DISCUSSION THREAD

**Discussion by xan105** [community/community#128436] [npm]
Similar to npm/feedback#122

Typescript 4.7 now supports the `exports` field
https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#package-json-exports-imports-and-self-referencing)
and adds a `types` import condition.

```json
{
    "name": "my-package",
    "type": "module",
    "exports": {
        ".": {
             "types": "./types/index.d.ts",
             "default": "./lib/index.js"
         },
         "./foo": {
             "types": "./types/foo.d.ts",
             "default": "./lib/foo.js"
         }  
     }
}

```
_Note that the "types" condition should always come first in "exports"._

Currently if a package only use the exports field with types the TS icon doesn't appear.
Will this be supported ?

**Comment by MylesBorins** [comment]
I've documented this internally as a bug on our tracker for engineering to pick up. Do you have any example packages

**Reply by samuelstroschein** [reply]
I seem to have the stated problem with the following package https://www.npmjs.com/package/@inlang/core
